src/tests/system/tests/test_schema.py

```diff
@@ -14,163 +19,51 @@
-    :setup:
-        1. Create new user "user1"
-        2. Add "description:gecos, userID:uidNumber, shell:loginShell, groupID:gidNumber" to ldap_user_extra_attrs
-        3. Add "ldap_id_mapping" to domain config, to ensure correct ids on all topologies
-        4. Start SSSD
-    :steps:
-        1. Run "getent passwd user1" to store user attributes to cache
-        2. Run ldbsearch command
-    :expectedresults:
-        1. User is found
-        2. Result has correct values
-    :customerscenario: True
-    """
-    provider.user("user1").add(gid=111111, uid=100110, gecos="gecos user1", shell="/bin/sh", home="/home/user1")
-    client.sssd.domain["ldap_user_extra_attrs"] = (
-        "description:gecos, userID:uidNumber, shell:loginShell, groupID:gidNumber"
-    )
-    client.sssd.domain["ldap_id_mapping"] = "false"
-    client.sssd.start()
-
     result = client.tools.getent.passwd("user1")
-    assert result is not None, "getent passwd user1 failed"
-
-    search = client.ldb.search(
-        f"/var/lib/sss/db/cache_{client.sssd.default_domain}.ldb", f"cn=users,cn={client.sssd.default_domain},cn=sysdb"
-    )
-
-    user_dict = search["name=user1@test,cn=users,cn=test,cn=sysdb"]
-    assert user_dict["description"] == ["gecos user1"], "attribute 'description' was not correct"
-    assert user_dict["shell"] == ["/bin/sh"], "attribute 'shell' was not correct"
-    assert user_dict["userID"] == ["100110"], "attribute 'userID' was not correct"
-    assert user_dict["groupID"] == ["111111"], "attribute 'groupID' was not correct"
-
-
-@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
-def test_schema__ldap_extra_attrs_negative_cache(client: Client, provider: GenericProvider):
-    """
-    :title: When extra attribute of user is added but not assigned, it is neither cached nor displayed
-    :setup:
-        1. Create new user "user1"
-        2. Add "number:telephonenumber" to ldap_user_extra_attrs
-        3. Start SSSD
-    :steps:
-        1. Run "getent passwd user1" to store user to cache
-        2. Run ldbsearch command
-    :expectedresults:
-        1. User is found
-        2. "number" is not in the output
-    :customerscenario: False
-    """
-    provider.user("user1").add()
-
-    client.sssd.domain["ldap_user_extra_attrs"] = "number:telephonenumber"
-
-    client.sssd.start()
-
-    result = client.tools.getent.passwd("user1")
-    assert result is not None, "User is not found"
-    assert result.name == "user1", "User has wrong name"
-
-    search = client.ldb.search(
-        f"/var/lib/sss/db/cache_{client.sssd.default_domain}.ldb", f"cn=users,cn={client.sssd.default_domain},cn=sysdb"
-    )
-
-    user_dict = search["name=user1@test,cn=users,cn=test,cn=sysdb"]
-    with pytest.raises(KeyError):
-        user_dict["number"]
-
-
-@pytest.mark.topology(KnownTopology.LDAP)
-def test_schema__ldap_extra_attrs_extra_email(client: Client, ldap: LDAP):
```

Comment:
Why is this test deleted? Checking that extra attributes configured in sssd (email in this case) are stored in ldb looks like reasonable scenario.

Reply:
This scenario is functionally tested. Under the goal that all tests are functional, black-box tests, as @aplopez suggested, this test shouldn't exist. We have yet to finalize a decision on it. I'm OK with keeping it. We have many of these tests, and I'd prefer to prune, improve, and have more comprehensive tests so we have less to main down the road. That's my only concern.

Reply:
We have tests that cover pieces that are not directly exposed to customers.
Issues in these areas will impact customers indirectly and will be hard to debug.
For example storing empty attributes will introduce bigger caches and decrease performance.
If extra attributes are not cached, they will be pulled from backend introducing performance impact as well.

Reply:
I don't disagree with you, but I don't think it falls under the scope of system tests. I honestly think we should create a /integration folder and put these kinds of test in there instead of system. I'll add it as a topic for Wednesday.

Reply:
@jakub-vavra-cz I added the tests back in, but under cache, with the integration marker